### PR TITLE
Codechange: Replace all uses of SmallMap with std::map

### PR DESCRIPTION
**Pull request by PeterN**
## Motivation / Problem

SmallMap.

## Description

Replace all uses of SmallMap with std::map. Some other adjustments were needed to make this possible:

1) Replacing a pointer with a unique_ptr in FontColourMap.
2) Swapping key/value around for SocketList due to const constraints (SocketList was modifying its keys), 

## Limitations

Given the changes to SocketList, this probably shouldn't be a map in the first place?

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/fios.h [hunk 1]
```diff
@@ -16,6 +16,7 @@
 #include "gamelog.h"
 #include "network/core/tcp_content_type.h"
 #include "timer/timer_game_calendar.h"
+#include <map>
```
Comment (TrueBrain) [inline src/fios.h]: I think we should add this to `stdafx.h`. As the fact it works for some platforms and not for others is just annoying, and we shouldn't want to deal with :)

Comment (PeterN) [reply src/fios.h]: Seems reasonable. We did it for vector already :)

**Comment by LordAro** [comment]
> [Codechange: Use std::map for gamelog's tempraroy grf name lookup.](https://github.com/OpenTTD/OpenTTD/pull/10836/commits/7e7b18f5bc18bf81eb32bfd5c198e934d0459ad2)

:)

File: src/fios.h [hunk 2]
```diff
@@ -26,7 +25,7 @@ enum SaveLoadInvalidateWindowData {
 	SLIWD_FILTER_CHANGES,        ///< The filename filter has changed (via the editbox)
 };
 
-typedef SmallMap<uint, CompanyProperties *> CompanyPropertiesMap;
+using CompanyPropertiesMap = std::map<uint, CompanyProperties *>;
```
Comment (rubidium42) [inline src/fios.h]: Rabbit hole-ish: `std::unique_ptr`?

**Review by rubidium42 — COMMENTED** [review]
I'd make the static initialisation and the flip of the socket list separate commits. The rest could be squashed into one commit I'd say.

File: src/fios.h (hunk 2 above)
Comment (PeterN) [reply src/fios.h]: Good point, I did it for the font map one already.

File: src/os/macosx/string_osx.cpp [hunk 3]
```diff
@@ -103,7 +103,7 @@ class CoreTextParagraphLayout : public ParagraphLayouter {
 				/* Extract font information for this run. */
 				CFRange chars = CTRunGetStringRange(run);
 				auto map = fontMapping.begin();
-				while (map < fontMapping.end() - 1 && map->first <= chars.location) map++;
+				while (std::next(map) != fontMapping.end() && map->first <= chars.location) map++;
```
Comment (rubidium42) [inline src/os/macosx/string_osx.cpp]: Wouldn't this be equal to `auto map = fontMapping.upper_bound(chars.location)`?

**Comment by PeterN** [comment]
Reorganized a bit. Fixes are done first, then conversion to map, then switching some things from raw pointer to unique_ptr.,

**Review by rubidium42 — APPROVED** [review]

**Pull request merged**